engine-workspace/src/contract.rs

```diff
@@ -0,0 +1,393 @@
+            amount.0,
+        ))
+    }
+
+    pub fn deploy_code(&self, code: Vec<u8>) -> CallDeployCode {
+        CallDeployCode::call(&self.contract).args(code)
+    }
+
+    pub fn deploy_erc20_token(&self, account_id: AccountId) -> CallDeployErc20Token {
+        CallDeployErc20Token::call(&self.contract).args_borsh(account_id)
+    }
+
+    pub fn call(&self, contract: Address, amount: U256, input: Vec<u8>) -> CallCall {
+        let value = WeiU256::from(amount);
+        let args = CallArgs::V2(FunctionCallArgsV2 {
+            contract,
+            value,
+            input,
+        });
+        CallCall::call(&self.contract).args_borsh(args)
+    }
+
+    pub fn submit(&self, input: Vec<u8>) -> CallSubmit {
+        CallSubmit::call(&self.contract).args(input)
+    }
+
+    pub fn register_relayer(&self, address: Address) -> CallRegisterRelayer {
+        CallRegisterRelayer::call(&self.contract).args_borsh(address)
+    }
+
+    pub fn ft_on_transfer(
+        &self,
+        sender_id: AccountId,
+        amount: U128,
+        message: String,
+    ) -> CallFtOnTransfer {
+        CallFtOnTransfer::call(&self.contract).args_json(json!({
+            "sender_id": sender_id,
+            "amount": amount,
+            "message": message
+        }))
+    }
+
+    pub fn factory_update(&self, bytes: Vec<u8>) -> CallFactoryUpdate {
+        CallFactoryUpdate::call(&self.contract).args(bytes)
+    }
+
+    pub fn fund_xcc_sub_account(
+        &self,
+        target: Address,
+        wnear_account_id: Option<AccountId>,
+    ) -> CallFundXccSubAccount {
+        let args = FundXccArgs {
+            target,
+            wnear_account_id,
+        };
+        CallFundXccSubAccount::call(&self.contract).args_borsh(args)
+    }
+
+    pub fn factory_set_wnear_address(&self, address: Address) -> CallFactorySetWNearAddress {
+        CallFactorySetWNearAddress::call(&self.contract).args_borsh(address)
+    }
+
+    pub fn stage_upgrade(&self, bytes: Vec<u8>) -> CallStageUpgrade {
+        CallStageUpgrade::call(&self.contract).args(bytes)
+    }
+
+    pub fn deploy_upgrade(&self) -> CallDeployUpgrade {
+        CallDeployUpgrade::call(&self.contract)
+    }
+
+    pub fn pause_precompiles(&self, paused_mask: u32) -> CallPausePrecompiles {
+        CallPausePrecompiles::call(&self.contract).args_borsh(paused_mask)
+    }
+
+    pub fn resume_precompiles(&self, paused_mask: u32) -> CallResumePrecompiles {
+        CallResumePrecompiles::call(&self.contract).args_borsh(paused_mask)
+    }
+
+    pub fn state_migration(&self) -> CallStateMigration {
+        CallStateMigration::call(&self.contract)
+    }
+
+    pub fn mint_account(
+        &self,
+        address: Address,
+        init_nonce: u64,
+        init_balance: u64,
+    ) -> CallMintAccount {
+        CallMintAccount::call(&self.contract).args_borsh((address, init_nonce, init_balance))
+    }
+
+    pub fn set_paused_flags(&self, flags: u8) -> CallSetPausedFlags {
```

Comment:
AFAIR we have a special type for it.

Reply:
For `CallSetPausedFlags`?

Reply:
Yes. We have an alias `pub type PausedMask = u8;`